drivers/SmartThings/zigbee-window-treatment/src/yoolax/init.lua

```diff
@@ -31,26 +35,35 @@ local function is_yoolax_window_shade(opts, driver, device)
   return false
 end
 
-local function set_shade_level(device, value, command)
+local function set_shade_level(driver, device, value, command)
   local level = 100 - value
   device:send_to_component(command.component, WindowCovering.server.commands.GoToLiftPercentage(device, level))
+  local timer = device.thread:call_with_delay(30, function ()
+    -- for some reason the device isn't updating us about its state so we'll send another bind request
+    device:send(device_management.build_bind_request(device, WindowCovering.ID, driver.environment_info.hub_zigbee_eui))
+    device:send(WindowCovering.attributes.CurrentPositionLiftPercentage:configure_reporting(device, 0, 600, 1))
+    device:send_to_component(command.component, WindowCovering.attributes.CurrentPositionLiftPercentage:read(device))
+  end)
+  device:set_field(LEVEL_UPDATE_TIMEOUT, timer)
 end
 
 local function window_shade_level_cmd(driver, device, command)
-  set_shade_level(device, command.value, command)
+  set_shade_level(driver, device, command.value, command)
 end
 
 local function window_shade_preset_cmd(driver, device, command)
-  set_shade_level(device, device.preferences.presetPosition, command)
+  set_shade_level(driver, device, device.preferences.presetPosition, command)
 end
 
 local function set_window_shade_level(level)
   return function(driver, device, cmd)
-    set_shade_level(device, level, cmd)
+    set_shade_level(driver, device, level, cmd)
   end
 end
 
 local function current_position_attr_handler(driver, device, value, zb_rx)
+  local timer = device:get_field(LEVEL_UPDATE_TIMEOUT)
+  if timer then device.thread:cancel_timer(timer) end
```

Comment:
I think you need to set timer to nil on the device object.

Reply:
that's a good catch